dts/bindings/qspi/st,stm32-qspi.yaml

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2020, Linaro limited
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    STM32 QSPI device representation. A stm32 quadspi node would typically
+    looks to this:
+
+        &quadspi {
+            pinctrl-0 = <&quadspi_clk_pe10 &quadspi_ncs_pe11
+                         &quadspi_bk1_io0_pe12 &quadspi_bk1_io1_pe13
+                         &quadspi_bk1_io2_pe14 &quadspi_bk1_io3_pe15>;
+
+            dmas = <&dma1 5 5 0x0000 0x03>;
+            dma-names = "tx_rx";
+
+            status = "okay";
+        };
+
+compatible: "st,stm32-qspi"
+
+include: base.yaml
+
+bus: qspi
```

Comment:
Does the hardware support dual or octal modes? The NXP controller supports single, dual, quad, and octal modes, so I've been debating whether to use `bus: qspi` like this in #31034.

Does it make sense that all the vendors have completely different dts definitions of a qspi bus, without inheriting from a common binding? Nothing analogous to `spi-controller.yaml` and `spi-device.yaml`? @galak, @pabigot ?

Reply:
This hardware block is dual and quad only, not octal.
I have no strong opinion on common binding, I don't know if there is much to share, as even bus definition may differ.